scripts/deploy.py

```diff
@@ -0,0 +1,123 @@
+#! /usr/bin/env python3
+import re, os, sys, subprocess
+
+# Define static variables
+CHAIN_ID            = '1'
+OPTIMIZER_ENABLED   = 'false'
+OPTIMIZER_RUNS      = '200'
+PATH_TO_SPELL       = 'src/DssSpell.sol'
+SPELL_CONTRACT_NAME = 'DssSpell'
+PATH_TO_EXEC_LIB    = './DssExecLib.address'
+SOLIDITY_USE        = 'solc:0.8.16'
+PATH_TO_CONFIG      = 'src/test/config.sol'
+
+# Read DssExecLib address
+with open(PATH_TO_EXEC_LIB, 'r') as f:
+    EXEC_LIB_ADDRESS = f.read().strip()
+if not EXEC_LIB_ADDRESS:
+    sys.exit(f'"{PATH_TO_EXEC_LIB}" file is empty or not found')
+
+# Check for uncommitted changes
+git_status = subprocess.run(['git', 'status', '--porcelain'], stdout=subprocess.PIPE, text=True, check=True).stdout.strip()
+if git_status:
+    sys.exit('There are uncommitted changes in the repository. Please commit or stash them before running this script')
+
+# Check env ETH_RPC_URL is set
+ETH_RPC_URL = os.environ.get('ETH_RPC_URL')
+if not ETH_RPC_URL:
+    sys.exit('Please set ETH_RPC_URL environment variable with RPC url')
+
+# Check ETH_RPC_URL is correct
+cast_chain_id = subprocess.run(['cast', 'chain-id'], stdout=subprocess.PIPE, text=True, check=True).stdout.strip()
+if cast_chain_id != CHAIN_ID:
+    sys.exit(f'Please provide correct ETH_RPC_URL. Currently set to chain id "{cast_chain_id}", expected "{CHAIN_ID}"')
+print(f'Using chain id {cast_chain_id}')
+
+# Check env ETHERSCAN_API_KEY is set
+ETHERSCAN_API_KEY = os.environ.get('ETHERSCAN_API_KEY')
+if not ETHERSCAN_API_KEY:
+    sys.exit('Please set ETHERSCAN_API_KEY environment variable')
+
+# Check env ETH_KEYSTORE is set
+ETH_KEYSTORE = os.environ.get('ETH_KEYSTORE')
+if not ETH_KEYSTORE:
+    # Use `cast wallet import --interactive "keystore_name"`
+    sys.exit('Please set ETH_KEYSTORE environment variable with path to the keystore')
+
+# Deploy the spell
+print('Deploying a spell...')
+deploy_logs = subprocess.run([
+    'forge', 'create',
+    '--no-cache',
+    '--broadcast',
+    '--optimize', OPTIMIZER_ENABLED,
+    '--optimizer-runs', OPTIMIZER_RUNS,
+    '--use', SOLIDITY_USE,
+    '--libraries', f'lib/dss-exec-lib/src/DssExecLib.sol:DssExecLib:{EXEC_LIB_ADDRESS}',
```

Comment:
Also `--libraries` could be moved to `foundry.toml`.
We could even get rid of the `DssExecLib.address` file.